Sort block

MayaGans opened the pull request:
- create sort block with descending checkbox
- delete old sort block
- delete descending block
- update documentation to reflect new block
- delete reverse block documentation

MayaGans committed:
sort block, remove reverse block

MayaGans committed:
sort documentation

MayaGans committed:
fix sort documentation

gvwilson commented on _posts/2019-08-26-values.md:
```diff
@@ -61,6 +61,21 @@ layout: null
+    <img src="{{'/guide/images/value_ifelse.png'| relative_url}}" width="100%">
+    </div>
+    <div id="second">
+    <h3>Description: The If - Then - Else block first checks its boolean condition. If the condition is true, the code held inside the Then space will activate, and the script will continue. If the condition is false, the code inside the Else space will activate.</h3>
```
Should probably say that this is done row-by-row - maybe a small HTML table showing an example?

gvwilson commented on _posts/2019-08-29-transform.md:
```diff
@@ -74,29 +74,16 @@ layout: null
     <h3>Arguments</h3>
-    <p>The name of the column(s) to arrange the rows by -- seperated by commas.</p>
+    <p><b>Text Field: </b>The name of the column(s) to arrange the rows by -- seperated by commas.</p>
+    <p><b>Checkbox: </b>Check to sort table in descending order</p>
```
Hm - just realized that there's no way to sort ascending by one column and descending by another, but how often does that ever come up?

gvwilson reviewed (approved):
Two minor questions - looks good.

MayaGans committed:
if then else doc

gvwilson reviewed (changes requested):
Please use `npm run test` to run the tests and update those as well to match these changes.

MayaGans committed:
Merge branch 'master' into sort-block

MayaGans committed:
sort block error handling

MayaGans commented:
VERY proud of this one !

- changed `sort` method to include `reverse` parameter (boolean for box checked)
- test_codegen for single column, multiple columns, multiple columns descending
- didn't touch test_exec (still works!)
- changed descending check box to `DESCENDING` to keep consistent

MayaGans committed:
fix error message to reflect descending

gvwilson commented on _posts/2019-08-26-values.md:
```diff
@@ -67,7 +67,9 @@ layout: null
     </div>
     <div id="second">
-    <h3>Description: The If - Then - Else block first checks its boolean condition. If the condition is true, the code held inside the Then space will activate, and the script will continue. If the condition is false, the code inside the Else space will activate.</h3>
+    <h3>Description: The If - Then - Else block first checks its boolean condition. 
```
Doesn't belong in this pull request (has nothing to do with sorting).

gvwilson commented on _posts/2019-08-29-transform.md:
```diff
@@ -74,29 +74,16 @@ layout: null
     <div id="second">
     <h3>Description:</h3>
-    <p>Order table rows by an expression involving its variables.</p>
+    <p>Order table rows by an expression involving its variables. Use the checkbox for descending the order of rows.</p>
```
"Use the checkbox to sort in descending order."

gvwilson commented on _posts/2019-08-29-transform.md:
```diff
@@ -74,29 +74,16 @@ layout: null
+    <p>Order table rows by an expression involving its variables. Use the checkbox for descending the order of rows.</p>
     <h3>Arguments</h3>
-    <p>The name of the column(s) to arrange the rows by -- seperated by commas.</p>
+    <p><b>Text Field: </b>The name of the column(s) to arrange the rows by -- seperated by commas.</p>
```
1. Use "," instead of "--".
2. "separated" not "seperated"

gvwilson commented on generators/js/transform_sort.js:
```diff
@@ -1,12 +1,16 @@
+
+
+const columns = block.getFieldValue('MULTIPLE_COLUMNS')
         .split(',')
```
we should probably write a helper function for this some day...

gvwilson commented on generators/js/transform_sort.js:
```diff
@@ -1,12 +1,16 @@
         .join(',')
-  return `.sort(${block.tbId}, [${columns}])`
+
+if (block.getFieldValue('DESCENDING') == 'FALSE') {
```
or:

```
const descending = (block.getFieldValue('DESCENDING') === 'FALSE')
return `.sort(${block.tbId}, [${columns}], ${descending})`
```

gvwilson commented on test/test_js_exec.js:
```diff
@@ -133,7 +114,8 @@ describe('execute blocks for entire pipelines', () => {
         'transform_sort',
-        {MULTIPLE_COLUMNS: 'red, green'})
+        {MULTIPLE_COLUMNS: 'red, green',
+        DESCENDING: 'FALSE'})
```
indent by one

gvwilson commented on test/test_js_error.js:
```diff
@@ -191,7 +191,8 @@ describe('raises errors at the right times', () => {
       makeBlock(
         'transform_sort',
-        {MULTIPLE_COLUMNS: ''})
+        {MULTIPLE_COLUMNS: ''},
```
this still looks wrong (two objects instead of one)

gvwilson commented on test/test_js_codegen.js:
```diff
@@ -206,16 +196,39 @@ describe('generate code for single blocks', () => {
+  const pipeline = makeBlock(
+    'transform_sort',
+    {MULTIPLE_COLUMNS: 'red,green',
+    DESCENDING: 'TRUE'})
```
indent by one so that the keys line up

gvwilson commented on test/test_js_codegen.js:
```diff
@@ -206,16 +196,39 @@ describe('generate code for single blocks', () => {
+    const pipeline = makeBlock(
+      'transform_sort',
+      {MULTIPLE_COLUMNS: 'blue',
+      DESCENDING: 'FALSE'})
```
indent by one so that the keys line up

gvwilson commented on tidyblocks/tidyblocks.js:
```diff
@@ -782,7 +773,12 @@ class TidyBlocksDataFrame {
       }, 0)
     })
-    return new TidyBlocksDataFrame(result)
+    if (reverse === false) {
```
this can more simply be written:

```
if (reverse) {
  result.reverse()
}
return new TidyBlocksDataFrame(result)
```

Remember, a Boolean can be used directly in a test because it *is* either `true` or `false` - any time you're comparing something with `(value === true)` or `(value === false)`, you can and should simplify.

gvwilson reviewed (changes requested):
Good work - small changes request.

MayaGans committed:
code cleanup

MayaGans committed:
revert if else block, add to a different pr

MayaGans committed:
resolving conflicts?
